client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/CustomEventLogListener.java

```diff
@@ -250,6 +287,7 @@ public CustomEvent convert(Integer key, String previousValue, Metadata previousM
       static class FilterConverter extends AbstractCacheEventFilterConverter<Integer, String, CustomEvent>
          implements Serializable {
          private final Object[] params;
+         private final CallbackCounter counter = new NumericCallbackCounter();
```

Comment:
This is not threadsafe and filters are supposed to be. I'm seeing spurious failures: testFilterCustomEventsInCluster(org.infinispan.client.hotrod.event.ClientClusterEventsTest)  and I suspect this to be the culprit.

Why not use an AtomicInteger here instead of reinventing it as CallbackCounter/NumericCallbackCounter?